plugins/destination/snowflake/client/client.go

```diff
@@ -43,7 +43,7 @@ func New(_ context.Context, logger zerolog.Logger, spec []byte, _ plugin.NewClie
 	if err != nil {
 		return nil, err
 	}
-	db, err := sql.Open("snowflake", dsn+"&BINARY_INPUT_FORMAT=BASE64&BINARY_OUTPUT_FORMAT=BASE64")
```

Comment:
What will happen if someone already has `timezone` configured in their DSN?

Reply:
Not sure, but can test it out

Reply:
If we store times with a timezone what's the reason we need it?

Reply:
Its a “usability” issue in that the default timezone is `America/Los_Angeles` (or whatever the user sets for the account or session)… This means that if a user compares `current_timestamp()` to `sync_time` (or any other timestamp synced by CQ) then they need to deal with casting and altering timezones of timestamps rather than just direct comparison. If we switch to to `timezone_tz` then all of the comparisons will just work out of the box

Reply:
> If we switch to to `timezone_tz` then all of the comparisons will just work out of the box

I agree we need to switch to `timezone_tz`. What's the reason we need to set the timezone in the DSN? If we save the timezone diff in the the column already

Reply:
The `datetime` object that we actually send to the DB has no timezone information associated with it as the Arrow Datetime object specifically documents that it has no timezone but can be assumed to be UTC

Reply:
I mean it shouldn't matter what's the timezone in the DSN as `timezone_tz` always save the data in diff from UTC format unless I misread the docs

Reply:
> The `datetime` object that we actually send to the DB has no timezone information associated with it as the Arrow Datetime object specifically documents that it has no timezone but can be assumed to be UTC

Shouldn't we convert everything to UTC before insert then? Or by setting UTC on the DSN it causes the Go driver to do it for us?

Reply:
Exactly, by setting the session Timezone Snowflake uses that timezone when there isn't one specified by the datetime object

Reply:
So we probably need to forbid setting it in the DSN? Or only set to UTC if not present so we still let users an option to configure this?

Reply:
I think we should forbid a user from setting any of the parameters we set in the DSN
`timezone`, `BINARY_INPUT_FORMAT`,  and`BINARY_OUTPUT_FORMAT`

Reply:
OK so we can do it as a follow up to not block this PR

Reply:
Issue to track the validation work: https://github.com/cloudquery/cloudquery/issues/17899